tests/test_config.py

```diff
@@ -265,50 +285,10 @@ def test_get_config_writing(patch_rootdir: None, monkeypatch: pytest.MonkeyPatch
     InvokeAIArgs.did_parse = False
 
 
-@pytest.mark.xfail(
-    reason="""
-    This test fails when run as part of the full test suite.
-
-    This test needs to deny nodes from being included in the InvocationsUnion by providing
-    an app configuration as a test fixture. Pytest executes all test files before running
-    tests, so the app configuration is already initialized by the time this test runs, and
-    the InvocationUnion is already created and the denied nodes are not omitted from it.
-
-    This test passes when `test_config.py` is tested in isolation.
-
-    Perhaps a solution would be to call `get_app_config().parse_args()` in
-    other test files?
-    """
-)
-def test_deny_nodes(patch_rootdir):
-    # Allow integer, string and float, but explicitly deny float
-    allow_deny_nodes_conf = OmegaConf.create(
-        """
-        InvokeAI:
-          Nodes:
-            allow_nodes:
-              - integer
-              - string
-              - float
-            deny_nodes:
-              - float
-        """
-    )
-    # must parse config before importing Graph, so its nodes union uses the config
-    get_config.cache_clear()
-    conf = get_config()
-    get_config.cache_clear()
-    conf.merge_from_file(conf=allow_deny_nodes_conf, argv=[])
-    from invokeai.app.services.shared.graph import Graph
-
-    # confirm graph validation fails when using denied node
-    Graph(nodes={"1": {"id": "1", "type": "integer"}})
-    Graph(nodes={"1": {"id": "1", "type": "string"}})
-
-    with pytest.raises(ValidationError):
-        Graph(nodes={"1": {"id": "1", "type": "float"}})
-
-    from invokeai.app.invocations.baseinvocation import BaseInvocation
+def test_deny_nodes():
+    config = get_config()
+    config.allow_nodes = ["integer", "string", "float"]
+    config.deny_nodes = ["float"]
```

Comment:
This will change the global config, lets instead instantiate a new config and let the test use that

Reply:
I was looking for a way to instantiate a new config and pass it to `get_invocations()` , but since this method gets its config from `get_config()` I couldn't figure out a non-hacky way of passing the custom config without altering the global.  My solution was to place a call to `get_config.cache_clear()` at the end of the test to avoid this, but I guess if the test fails halfway it will contaminate the global config. Perhaps a try/finally or context manager would work?

Reply:
Yeah, that would work - but also I think we could just instantiate a fresh config:

```py
config = DefaultInvokeAIAppConfig()
config.allow_nodes = ["integer", "string", "float"]
config.deny_nodes = ["float"]
```

Reply:
Sorry, you are correct. I think we need to use clear_cache.